spring-integration-mail/src/main/java/org/springframework/integration/mail/AbstractMailReceiver.java

```diff
@@ -448,5 +452,14 @@ public Folder getFolder() {
 			}
 		}
 
+		@Override
+		public Date getReceivedDate() throws MessagingException {
+			/*
+			 * Basic MimeMessage always returns null; delegate to the original.
+			 */
+			return source.getReceivedDate();
```

Comment:
Looks like we should do the same and for this too:
````java
public int getLineCount() throws MessagingException {
	return -1;
}
````

Let me know and I fix that on merge.